articles/securing-django-api/index.md

```diff
@@ -0,0 +1,251 @@
+    queryset = Todo.objects.all()
+    serializer_class = TodoSerializer
+
+
+class DeleteTodoAPIView(DestroyAPIView):
+    """This endpoint allows for deletion of a specific Todo from the database"""
+    permission_classes = (IsAuthenticated,)#permission classes
+    queryset = Todo.objects.all()
+    serializer_class = TodoSerializer
+
+```
+
+By adding permission classes to our view classes makes our API endpoints secure. When we visit `http://127.0.0.1:8000/api/v1/todo/` we get HTTP 403 Forbidden error.
+
+### Implementing the Token Authentication
+
+In `settings.py` file in our `django_todo` projects directory add `rest_framework.authtoken` to the `INSTALLED_APPS` list and also include `TokenAuthentication` to the `REST_FRAMEWORK` dictionary.
+
+```python
+INSTALLED_APPS = [
+    'django.contrib.admin',
+    'django.contrib.auth',
+    'django.contrib.contenttypes',
+    'django.contrib.sessions',
+    'django.contrib.messages',
+    'django.contrib.staticfiles',
+    'todo',
+    'rest_framework',
+    'rest_framework.authtoken',# <-- authtoken
+    'coreapi',
+]
+
+REST_FRAMEWORK = {
+    'DEFAULT_SCHEMA_CLASS': 'rest_framework.schemas.coreapi.AutoSchema',
+    'DEFAULT_AUTHENTICATION_CLASSES': [
+        'rest_framework.authentication.TokenAuthentication',  # <-- TokenAuthentication
+    ],
+}
+```
+Now we should run the command `./manage.py migrate` to create the table that will store the authentication tokens.
+
+```bash
+$ ./manage.py migrate
+```
+
+To create an authentication token, we need a user account that we can create through the command line.
+Run the command below to create a user account:
+```bash
+$ ./manage.py createsuperuser --username paul --email paul@domain.com
+```
+
+We can also generate the authentication token through the command line. Run the command `python manage.py drf_create_token paul` to generate the authentication token.
+```bash
+$ ./manage.py drf_create_token paul
+Generated token 342b58233e5fdeb2446bcaae60b6e51e953f7a17 for user paul
+```
+
+We will be using the generated token `342b58233e5fdeb2446bcaae60b6e51e953f7a17` to authenticate our requests to the server.
+
+Let's make a request to `http://127.0.0.1:8000/api/v1/todo/` adding our token as an authorization header `Authorization: Token 342b58233e5fdeb2446bcaae60b6e51e953f7a17`.
+![GET Request with token](/engineering-education/securing-django-api/get-using-token.png)
+
+### User token endpoint
+Django rest framework comes with an endpoint that users can use to generate their authentication tokens by providing their valid username and password.
+
+In the `django_todo` project directory add the api endpoint for token generation in the `urls.py` file.
+```python
+urlpatterns = [
+    path('admin/', admin.site.urls),
+    path('api/v1/todo/', include("todo.urls")),
+    path('docs/', include_docs_urls(title='Todo Api')),
+    path('api/token', obtain_auth_token, name="auth_toke"),#token generation url
```

Comment:
where have you defined obtain_auth_token? I am getting an error at this step! Can you check this out?